spec/lib/site_setting_extension_spec.rb

```diff
@@ -1048,6 +1077,43 @@ def self.translate_names?
       expect(SiteSetting.search_experience(theme_id: theme_2.id)).to eq("search_field")
     end
 
+    it "publishes changes across processes" do
+      MessageBus.on
+
+      settings_tss_instance_1 = new_settings(provider_local)
+      settings_tss_instance_2 = new_settings(provider_local)
+
+      settings_tss_instance_1.listen_for_changes = true
+      settings_tss_instance_2.listen_for_changes = true
+
+      settings_tss_instance_1.load_settings(File.join(Rails.root, "config", "site_settings.yml"))
+      settings_tss_instance_2.load_settings(File.join(Rails.root, "config", "site_settings.yml"))
+
+      settings_tss_instance_1.refresh!
+      settings_tss_instance_2.refresh!
+
+      expect(settings_tss_instance_1.enable_welcome_banner(theme_id: theme_1.id)).to eq(false)
+      expect(settings_tss_instance_2.enable_welcome_banner(theme_id: theme_1.id)).to eq(false)
+
+      result =
+        Themes::ThemeSiteSettingManager.call(
```

Comment:
I don't think this service should be the concern of this file which is meant to hold unit tests for methods of the `SiteSettingExtension` module. I would personally call the `SiteSetting.change_themeable_site_setting` directly here.

Reply:
I had SiteSetting.change_themeable_site_setting originally but I did this because I need to update the value too...but yeah I could go around the service and set the DB value directly then call SiteSetting.change_themeable_site_setting , will change